server/session.go

```diff
@@ -0,0 +1,273 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package server
+
+import (
+	"errors"
+	"sync"
+
+	pb "github.com/googleapis/gapic-showcase/server/genproto"
+)
+
+type Session interface {
+	GetName() string
+	GetReport() *pb.ReportSessionResponse
+	AddTest(t Test) error
+	DeleteTest(name string) error
+	AddAnswers(name string, answers []string) error
+	TestAnswers(name string, answers []string) error
+}
+
+var instance Session
+var once sync.Once
+
+func GetSessionSingleton() Session {
+	once.Do(func() {
+		instance = &sessionImpl{
+			name:  "-",
+			mu:    &sync.Mutex{},
+			tests: map[string]Test{},
+		}
+	})
+	return instance
+}
+
+type sessionImpl struct {
+	name string
+
+	mu    *sync.Mutex
+	tests map[string]Test
+}
+
+func (s *sessionImpl) GetName() string {
+	return s.name
+}
+
+func (s *sessionImpl) GetReport() *pb.ReportSessionResponse {
+	numRequired := 0
```

Comment:
I'd wrap it like this:
```
var (
numRequired int
numRequiredSkipped int
//...
errors []*pb.ResportSessionResponse_Issue // no need to init array, append already knows what to do
)
```

Also consider creating a type `type result struct {test, skipped, failed int}`

Reply:
Done.